build.psm1

```diff
@@ -99,18 +99,16 @@ function Get-PSCommitId
 
 function Get-EnvironmentInformation
 {
-    $environment = @{}
+    $environment = @{'IsWindows' = [System.Environment]::OSVersion.Platform -eq [System.PlatformID]::Win32NT}
     # PowerShell will likely not be built on pre-1709 nanoserver
-    if ($PSVersionTable.ContainsKey("PSEdition") -and "Core" -eq $PSVersionTable.PSEdition) {
-        $environment += @{'IsCoreCLR' = $true}
-        $environment += @{'IsLinux' = $IsLinux}
-        $environment += @{'IsMacOS' = $IsMacOS}
-        $environment += @{'IsWindows' = $IsWindows}
+    if ('System.Management.Automation.Platform' -as [type]) {
```

Comment:
Has this been tested on Windows PowerShell?

Reply:
I believe `PowerShell-CI-windows` build runs on Windows PowerShell
```
2020-01-13T20:20:50.2380725Z ========================== Starting Command Output ===========================
2020-01-13T20:20:50.2632322Z ##[command]"C:\windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'D:\a\_temp\a270093b-9d0f-4692-abef-11227601b6c1.ps1'"
```
https://dev.azure.com/powershell/2972bb5c-f20c-4a60-8bd9-00ffe9987edc/_apis/build/builds/43352/logs/16

Reply:
And I also tested locally.